pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -125,7 +125,8 @@ func SetDefaults_Shoot(obj *Shoot) {
 		obj.Spec.Kubernetes.KubeControllerManager = &KubeControllerManagerConfig{}
 	}
 
-	if obj.Spec.Kubernetes.EnableStaticTokenKubeconfig == nil {
+	k8sLess127, _ := versionutils.CheckVersionMeetsConstraint(obj.Spec.Kubernetes.Version, "< 1.27")
+	if obj.Spec.Kubernetes.EnableStaticTokenKubeconfig == nil && k8sLess127 {
```

Comment:
I think there are many places in the code that assume that `EnableStaticTokenKubeconfig==nil` means "enabled" (like the warning that you handled explicitly). Instead of putting `if k8s < 1.27` conditions everywhere, perhaps it's better to simply default the value to `false` even for `Shoot`s > `1.27`?